cfme/tests/services/test_catalog_item.py

```diff
@@ -27,33 +31,54 @@ def dialog():
         default_text_box="default value"
     )
 
-    service_dialog = ServiceDialog(label=dialog, description="my dialog",
+    service_dialog = ServiceDialog(label=dialog_name, description="my dialog",
                                    submit=True, cancel=True,
                                    tab_label="tab_" + fauxfactory.gen_alphanumeric(),
                                    tab_desc="my tab desc",
                                    box_label="box_" + fauxfactory.gen_alphanumeric(),
                                    box_desc="my box desc")
     service_dialog.create(element_data)
-    flash.assert_success_message('Dialog "{}" was added'.format(dialog))
-    yield dialog
+    flash.assert_success_message('Dialog "{}" was added'.format(dialog_name))
+    yield service_dialog
 
+    # fixture cleanup
+    try:
+        service_dialog.delete()
+    except NoSuchElementException:
+        logger.warning('test_catalog_item: dialog yield fixture cleanup, dialog "{}" not '
+                       'found'.format(dialog_name))
 
-@pytest.yield_fixture(scope="function")
+
+@pytest.yield_fixture(scope="module")
 def catalog():
-    catalog = "cat_" + fauxfactory.gen_alphanumeric()
-    cat = Catalog(name=catalog,
+    catalog_name = "test_cat_" + fauxfactory.gen_alphanumeric()
+    cat = Catalog(name=catalog_name,
                   description="my catalog")
     cat.create()
-    yield catalog
+    yield cat
+
+    # fixture cleanup
+    try:
+        cat.delete()
+    except NoSuchElementException:
+        logger.warning('test_catalog_item: catalog yield fixture cleanup, catalog "{}" not '
+                       'found'.format(catalog_name))
 
 
 @pytest.yield_fixture(scope="function")
 def catalog_item(dialog, catalog):
-    catalog_item = CatalogItem(item_type="Generic",
-                               name=fauxfactory.gen_alphanumeric(),
-                               description="my catalog", display_in=True,
-                               catalog=catalog, dialog=dialog)
-    yield catalog_item
+    cat_item = CatalogItem(item_type="Generic",
+                           name='test_item_' + fauxfactory.gen_alphanumeric(),
+                           description="my catalog item", display_in=True,
+                           catalog=catalog.name, dialog=dialog.label)
```

Comment:
Changed the dialog and catalog fixtures to return the object, so here we've got to access their attributes.